internal/resources/resources.go

```diff
@@ -1,25 +1,49 @@
-package hcsoci
+package resources
 
 import (
 	"context"
 	"errors"
-	"os"
 
+	"github.com/Microsoft/hcsshim/internal/credentials"
+	"github.com/Microsoft/hcsshim/internal/layers"
 	"github.com/Microsoft/hcsshim/internal/log"
-	"github.com/Microsoft/hcsshim/internal/ospath"
 	"github.com/Microsoft/hcsshim/internal/uvm"
 )
 
-const (
-	scratchPath = "scratch"
-	rootfsPath  = "rootfs"
-)
-
 // NetNS returns the network namespace for the container
 func (r *Resources) NetNS() string {
 	return r.netNS
 }
 
+func (r *Resources) SetNetNS(netNS string) {
+	r.netNS = netNS
+}
+
+func (r *Resources) SetCreatedNetNS(created bool) {
+	r.createdNetNS = true
+}
+
+// ContainerRootInUVM returns the aontainerRootInUVM for the container
+func (r *Resources) ContainerRootInUVM() string {
+	return r.containerRootInUVM
+}
+
+func (r *Resources) SetContainerRootInUVM(containerRootInUVM string) {
+	r.containerRootInUVM = containerRootInUVM
+}
+
+func (r *Resources) SetAddedNetNSToVM(addedNetNSToVM bool) {
+	r.addedNetNSToVM = addedNetNSToVM
+}
+
+func (r *Resources) SetLayers(l *layers.ImageLayers) {
+	r.layers = l
+}
+
+func (r *Resources) Add(newResource ...ResourceCloser) {
```

Comment:
Do we envision adding more than one resource at a time in the future? If so newResource should be named newResources. Also add a brief comment to the tune of "Add adds one or more resources to be tracked for release later on blah blah....."

Reply:
Yes, there's a use case for that in my hyper-v assigned devices work. I've added a comment and updated.